src/theory/quantifiers/equality_query.cpp

```diff
@@ -111,74 +111,74 @@ Node EqualityQueryQuantifiersEngine::getInternalRepresentative(Node a,
       }
     }
   }
+  TypeNode v_tn = q.isNull() ? a.getType() : q[0][index].getType();
   if (options::quantRepMode() == options::QuantRepMode::EE)
   {
-    return r;
+    int score = getRepScore(r, q, index, v_tn);
+    if (score >= 0)
+    {
+      // if we are not a valid representative, try to select one below
```

Comment:
it's a bit confusing that you say this but then exits the method. Please rephrase to make clear you are referring to the negative case, when you did not enter the `if`